allennlp/data/dataset_readers/coreference_resolution/winobias.py

```diff
@@ -0,0 +1,167 @@
+                    inside_square_span = True
+                elif inside_square_span or "]" in token:
+                    old_span = clusters[0][-1]
+                    clusters[0][-1] = (old_span[0], old_span[1] + 1)
+                    inside_square_span = False
+
+                if "(" in token and ")" in token:
+                    clusters[1].append((index, index))
+                elif "(" in token:
+                    clusters[1].append((index, index))
+                    inside_round_span = True
+                elif inside_round_span or ")" in token:
+                    old_span = clusters[1][-1]
+                    clusters[1][-1] = (old_span[0], old_span[1] + 1)
+                    inside_round_span = False
+
+                if token.endswith("."):
+                    # Winobias is tokenised, but not for full stops.
+                    # We'll just special case them here.
+                    token = token[:-1]
+                    words.append(token.strip("[]()"))
+                    words.append(".")
+                else:
+                    words.append(token.strip("[]()"))
+
+            yield self.text_to_instance(words, [x for x in clusters.values()])
+
+    @overrides
+    def text_to_instance(self,  # type: ignore
+                         sentence: List[str],
+                         gold_clusters: Optional[List[List[Tuple[int, int]]]] = None) -> Instance:
+        # pylint: disable=arguments-differ
+        """
+        Parameters
+        ----------
+        sentences : ``List[str]``, required.
+            The sentence to analyse.
+        gold_clusters : ``Optional[List[List[Tuple[int, int]]]]``, optional (default = None)
+            A list of all clusters in the sentence, represented as word spans. Each cluster
+            contains some number of spans, which can be nested and overlap, but will never
+            exactly match between clusters.
+
+        Returns
+        -------
+        An ``Instance`` containing the following ``Fields``:
+            text : ``TextField``
+                The text of the full sentence.
+            spans : ``ListField[SpanField]``
+                A ListField containing the spans represented as ``SpanFields``
+                with respect to the sentence text.
+            span_labels : ``SequenceLabelField``, optional
+                The id of the cluster which each possible span belongs to, or -1 if it does
+                 not belong to a cluster. As these labels have variable length (it depends on
+                 how many spans we are considering), we represent this a as a ``SequenceLabelField``
+                 with respect to the ``spans ``ListField``.
+        """
+        metadata: Dict[str, Any] = {"original_text": sentence}
+        if gold_clusters is not None:
+            metadata["clusters"] = gold_clusters
+
+        text_field = TextField([Token(word) for word in sentence], self._token_indexers)
```

Comment:
This probably won't interact well with a `Predictor`, which will probably use a tokenizer to process input.  You might want to have the conversion to `Tokens` in `_read`.